### PR TITLE
[Feature] Support functions sinh, cosh and tanh

### LINKED ISSUE #13298: [function] sr already support bunch of trigonometric functions, still need sinh, cosh, tanh
trigonometric functions
sinh
cosh 
tanh

image

### PR DESCRIPTION
**Pull request by tomscut**
Signed-off-by: Tao Li <tomscut@apache.org>

## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：
Fixes #13298

## Problem Summary(Required) ：
Support trigonometric functions `sinh`, `cosh` and `tanh`.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
- [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/test/exprs/vectorized/math_functions_test.cpp [hunk 1]
```diff
@@ -1308,6 +1308,39 @@ TEST_F(VecMathFunctionsTest, OutputNanTest) {
...
+    }
+
+    {
+        std::vector<bool> null_expect = {false, false, true};
+        std::unique_ptr<FunctionContext> ctx(FunctionContext::create_test_context());
+        ColumnPtr result = MathFunctions::cosh(ctx.get(), columns);
+        auto nullable = ColumnHelper::as_raw_column<NullableColumn>(result);
+        ASSERT_EQ(nullable->size(), null_expect.size());
+        for (size_t i = 0; i < nullable->size(); i++) {
+            ASSERT_EQ(nullable->is_null(i), null_expect[i]);
```
Comment (imay) [inline be/test/exprs/vectorized/math_functions_test.cpp:1329]: Besides the values of null, the result should be checked too.

Comment (tomscut) [reply be/test/exprs/vectorized/math_functions_test.cpp:1329]: Thanks @imay for your review. I will add the check.

Comment (tomscut) [reply be/test/exprs/vectorized/math_functions_test.cpp:1376]: Hi @imay , I add TrigonometricFunctionTest to check the results. Please have a look. Thank you.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by murphyatwork — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**